src/systems/filecoin_blockchain/storage_power_consensus/storage_power_actor_code.go

```diff
@@ -192,10 +134,12 @@ func (a *StoragePowerActorCode_I) RemoveStorageMiner(rt Runtime) {
 }
 
 func (a *StoragePowerActorCode_I) EnsurePledgeCollateralSatisfied(rt Runtime) {
-	rt.ValidateImmediateCallerAcceptAnyOfType(actor.BuiltinActorID_StorageMiner)
 
-	TODO() // TODO: principal access control
 	minerAddr := rt.ImmediateCaller()
+	ok := rt.ValidateIsStorageMiner(minerAddr)
+	if !ok {
+		rt.AbortStateMsg("spa.EnsurePledgeCollateralSatisfied: caller must be StorageMiner.")
+	}
```

Comment:
Unnecessary! If the caller has an entry in the collateral table, they they're fine.